Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/q.d.ts" />
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+var adal = require ('adal-node');
+
+import taskLib = require ('vsts-task-lib/task');
+var request = require ('request');
+import Q = require ('q');
+var parseString = require('xml2js').parseString;
+
+var AuthenticationContext = adal.AuthenticationContext;
+
+function getAuthorizationToken ( SPN ) : Q.Promise <string> {
+
+	var deferred = Q.defer<string>();
+	var authorityUrl = 'https://login.windows.net/' + SPN.tenantID; 
```

Comment:
there are several azure URLs spread across the file. It will be good if we can gather them as constants in one place.